### PR TITLE
chore: add additional examples and TODOs for issues

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -9,4 +9,4 @@ module(
 bazel_dep(name = "bazel_skylib", version = "1.1.1")
 bazel_dep(name = "rules_nodejs", version = "5.5.0")
 bazel_dep(name = "aspect_rules_js", version = "0.13.0")
-bazel_dep(name = "aspect_bazel_lib", version = "1.3.1")
+bazel_dep(name = "aspect_bazel_lib", version = "1.7.0")
```
Comment (jbedard) [inline MODULE.bazel:12]: Would we prefer putting the updates in a separate commit? Personally I'd rather not have runtime changes in an examples commit 🤷

File: examples/deps_lib/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    name = "tsconfig",
+    src = "tsconfig.json",
+    visibility = [":__subpackages__"],
+)
+
+ts_project(
+    name = "deps_ts",
+    srcs = ["index.ts"],
+    declaration = True,
+    # TODO: fix worker mode flaky bug when there are multiple ts_project targets in a package
```
Comment (jbedard) [inline examples/deps_lib/BUILD.bazel:14]: Worth creating an issue?

File: examples/deps_lib/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        "//examples:node_modules/date-fns",
+        "//examples/deps_lib/b",
+    ],
+)
+
+npm_package(
+    name = "deps_lib",
+    srcs = [
+        ":deps_ts",
+        # TODO: This seems undesirable to have to specify the transitive closure of ts_project
```
Comment (jbedard) [inline examples/deps_lib/BUILD.bazel:26]: Is this different then https://github.com/aspect-build/rules_ts/issues/96?

**Review by jbedard — APPROVED** [review]

File: examples/BUILD.bazel [hunk 4]
```diff
@@ -17,5 +17,14 @@ npm_link_package(
     visibility = ["//examples:__subpackages__"],
 )
 
+npm_link_package(
+    name = "node_modules/@myorg/deps_lib",
+    src = "//examples/deps_lib",
+    root_package = "examples",
+    visibility = ["//examples:__subpackages__"],
+    # TODO: propagate this dependency from //examples/deps_lib: https://github.com/aspect-build/rules_ts/issues/96
+    deps = {":.aspect_rules_js/node_modules/date-fns/2.29.1": ""},
```
Comment (thesayyn) [inline examples/BUILD.bazel:26]: nit: this might be hard for users to understand. is there an alias for this that is more readable?

**Review by thesayyn — APPROVED** [review]

**Pull request merged**